src/main/java/console/ConsoleReader.java

```diff
@@ -0,0 +1,39 @@
+package console;
+
+import exceptions.InputStreamException;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+public class ConsoleReader {
+
+    private final BufferedReader bufferedReader;
+
+    public ConsoleReader(InputStream inputStream) {
+        bufferedReader = new BufferedReader(new InputStreamReader(checkedStream(inputStream)));
+    }
+
+    public String readUserInput() {
+        String userInput;
+        try {
+            userInput = bufferedReader.readLine();
+            if (userInput == null) {
+                userInput = "#quit";
+            }
+        } catch (IOException e) {
+            throw new InputStreamException(e.getMessage());
+        }
+        return userInput;
+    }
+
+    public InputStream checkedStream(InputStream inputStream) {
```

Comment:
I don't think this method does what you intend it to do 😸 

I don't think there is a way `return inputStream` would ever cause an NPE.
Is there a test for this that demonstrates the behaviour you want to see?